presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSplitSource.java

```diff
@@ -249,6 +253,69 @@ public void testEmptyBucket()
         assertEquals(getSplits(hiveSplitSource, OptionalInt.of(3), 10).size(), 0);
     }
 
+    @Test
+    public void testPreloadSplitsForGroupedExecution()
+            throws Exception
+    {
+        // TODO: Use Session::builder after HiveTestUtils::SESSION is refactored to Session.
+        ConnectorSession session = new TestingConnectorSession(
+                new HiveSessionProperties(
+                        new HiveClientConfig().setPreloadSplitsForGroupedExecution(true),
+                        new OrcFileWriterConfig(),
+                        new ParquetFileWriterConfig()).getSessionProperties());
+        HiveSplitSource hiveSplitSource = HiveSplitSource.bucketed(
+                session,
+                "database",
+                "table",
+                TupleDomain.all(),
+                10,
+                10,
+                new DataSize(1, MEGABYTE),
+                new TestingHiveSplitLoader(),
+                Executors.newFixedThreadPool(5),
+                new CounterStat());
+        for (int i = 0; i < 10; i++) {
+            hiveSplitSource.addToQueue(new TestSplit(i, OptionalInt.of(0)));
+            assertEquals(hiveSplitSource.getBufferedInternalSplitCount(), i + 1);
+        }
+
+        final SettableFuture<List<ConnectorSplit>> splits = SettableFuture.create();
```

Comment:
nit: is this `final` required? IIRC lambda capture works with "effectively final"

I noted `testReaderWaitsForSplits()` also mark it as `final`. @nezihyigitbasi : Do you think the modifier `final` is necessary here?

Reply:
Looks unnecessary to me too. Same for `started` variable below.